hydrosdk/application.py

```diff
@@ -126,6 +130,31 @@ def update_status(self):
         Poll a cluster for a new Application status.
         """
         self.status = self.find(self.cluster, self.name).status
+    
+    def lock_till_available(self, timeout: int = 120) -> 'Application':
```

Comment:
Minor comment - wouldn't `lock_till_ready` be a clearer naming? Since we are waiting for `ApplicationStatus.READY`

Reply:
Logically we wait for the application to become available for serving. We don't really care what are the internal statuses for an entity. So, I thought it would be a good idea to generalize this. The same applies to servables.

Reply:
Since there are two conditions for terminating the lock, maybe it should be something like `lock_while_creating` or `lock_while_starting`. Also, we don't use other statuses, so there is no need to name methods with exact enum items.